routes/projects/create.js

```diff
@@ -0,0 +1,28 @@
+/*
+ * POST /projects
+ * Creates a new project for the given user. A project object is sent in
+ * the form:
+ * {
+ *   name: <string> required,
+ *   username: <string> required,
+ *   isPublic: <boolean> required,
+ *   publishedAt: <dataTime> optional,
+ *   publishLink: <string> optional,
```

Comment:
The client won't send `publishedAt` or `publishLink`, so I think this is confusing.  It will get *stored* with this data, but I don't think you want to require it to be *sent* in this form.

I think there are optional bits to this.  For example:

```js
{
  name: "...",
  username: "...",
  files: [initial, list, of, files]
}
```

In this case, `isPublic` can take the value `false` since it is absent, and `publishedAt` and `publishLink` will get updated later.